### ISSUE TITLE
Unnecessary parenthesis completely changes meaning if removed

### ISSUE THREAD

**Issue by Bilge** [kalessil/phpinspectionsea#394]
Given the following code:

```php
if (!($first = \iter\take(1, $values))->valid())
```

PHPEA claims the inner braces are redundant and thus changes the code to the following:

```php
if (!$first = \iter\take(1, $values)->valid())
```

This changes the type of `$first` from `Iterator` to `bool`. Since these two statements are non-equivalent this must be considered a bug in the inspection.

**Comment by kalessil** [comment]
Good catch @Bilge, I'll take care of the bug.
